packages/router-plugin/src/router.module.ts

```diff
@@ -4,15 +4,14 @@ import { NgxsModule } from '@ngxs/store';
 import { RouterState } from './router.state';
 import { DefaultRouterStateSerializer, RouterStateSerializer } from './serializer';
 
-@NgModule()
+@NgModule({
+  imports: [NgxsModule.forFeature([RouterState])]
+})
 export class NgxsRouterPluginModule {
-  static forRoot(): ModuleWithProviders[] {
-    return [
-      NgxsModule.forFeature([RouterState]),
-      {
-        ngModule: NgxsRouterPluginModule,
-        providers: [{ provide: RouterStateSerializer, useClass: DefaultRouterStateSerializer }]
-      }
-    ];
+  static forRoot(): ModuleWithProviders {
+    return {
+      ngModule: NgxsRouterPluginModule,
+      providers: [{ provide: RouterStateSerializer, useClass: DefaultRouterStateSerializer }]
+    };
```

Comment:
I'm trying to figure out what fixes the issue?
Was it moving the `NgxsModule.forFeature` to the top?

Reply:
Ivy automatically infer type `ModuleWithProviders<T>` by type from `ngModule`

```ts
return {
      ngModule: NgxsRouterPluginModule,
      providers: [{ provide: RouterStateSerializer, useClass: DefaultRouterStateSerializer }]
};
```

Also by Angular Guide forRoot need return `ModuleWithProviders` instead `ModuleWithProviders[]`

Reply:
ah, ok. Thanks!

Reply:
I wonder if there are any other plugins we need to fix this for?
... any in the labs? @joaqcid maybe your `actions-executing` because it also has its own state.

Reply:
@markwhitfeld https://github.com/ngxs-labs/actions-executing/blob/master/src/lib/actions-executing.module.ts

Reply:
Not sure if any of the others hold their own state.

Reply:
> Not sure if any of the others hold their own state.

What do you think?